taxonomy.html

```diff
@@ -0,0 +1,478 @@
+        existing ones.
+      </p>
+      <section>
+        <h3>
+          Terms
+        </h3>
+        <p>
+          An <dfn data-lt="accessible taxonomies">accessible taxonomy</dfn> or
+          <dfn data-lt="taxonomies">taxonomy</dfn> of an <a>accessible
+          property</a> like <a>role</a> or <a>state</a> is a hierarchy of all
+          possible values of the <a>accessible property</a>. The taxonomy
+          describes each value by set of attributes, for example, it may
+          provide a human readable localized string for the property value.
+          Also the taxonomy describes interconnection between values from other
+          taxonomies, for example, <code>link</code> role may be connected to
+          <code>visited</code> state.
+        </p>
+      </section>
+      <section class="informative">
+        <h3>
+          Understanding the terms.
+        </h3>
+        <p>
+          Each <a>accessible element</a> is described by accessible properties
+          like <a>role</a>, <a>state</a>, <a>actions</a> etc. Every accessible
+          property may take a range of values, for example,
+          <code>checkbox</code> and <code>paragraph</code> as values for
+          <a>role</a>. All accessible properties may have various dependencies,
+          for example, <code>checked</code> state can be applied to
+          <code>checkbox</code> role, however it's not applicable to
+          <code>paragraph</code> role. Accessible property values and their
+          dependencies are structured into <a>taxonomies</a>. The <a>assistive
+          technology</a> can operate with them.
+        </p>
+        <p>
+          The values of accessible properties and their relations are specific
+          to a markup whose semantics they serve to express. If the web author
+          needs to expose the semantics of some new kind, for example, to
+          describe a music sheet, then he may need to introduce new roles,
+          states etc and describe them. To do so the web author needs to extend
+          existing <a>taxonomies</a>. The <a>assistive technology</a> will able
+          then to understand the semantics of the content and figure out a
+          connection between 'known' taxa. For example, if the web author
+          introduces <code>checklistitem</code> role which is compound from two
+          <code>checkbox</code> and <code>listitem</code> roles, i.e it
+          inherits properties of both of them then the web author should
+          integrate the new role into existing role hierarchy.
+        </p>
+      </section>
+    </section>
+    <section>
+      <h2>
+        Extension to the <code>AccessibleDocument</code> interface
+      </h2>
+      <p>
+        An <a>accessible document</a> provides a method to import taxonomies.
+      </p>
+      <pre class="idl">
+partial interface AccessibleDocument {
+  Taxon? taxonOf(DOMString taxonomy, DOMString name);
+  [ Throws ]
```

Comment:
Hi, Marcos. It seems there's some confusion about branches. I created two
branches and pull requests, one is for patterns, another one is for
taxonomies. The patterns one was based on taxonomies, and thus you reviewed
it as part of patterns branch.

If I do the work in parallel, then what is a better way to avoid confusion?
Should each branch be positioned against master gh-pages? What is a best
analogue of MQ in git where I was able to stack patches on top of each
other?

On Thu, Nov 19, 2015 at 9:54 PM, Marcos Caceres <notifications@github.com>
wrote:

> In taxonomy.html
> <https://github.com/WICG/a11yapi/pull/28#discussion_r45431281>:
>
> > +          inherits properties of both of them then the web author should
> > +          integrate the new role into existing role hierarchy.
> > +        </p>
> > +      </section>
> > +    </section>
> > +    <section>
> > +      <h2>
> > +        Extension to the <code>AccessibleDocument</code> interface
> > +      </h2>
> > +      <p>
> > +        An <a>accessible document</a> provides a method to import taxonomies.
> > +      </p>
> > +      <pre class="idl">
> > +partial interface AccessibleDocument {
> > +  Taxon? taxonOf(DOMString taxonomy, DOMString name);
> > +  [ Throws ]
>
> Throws is only valid in Gecko webIDL :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WICG/a11yapi/pull/28/files#r45431281>.
>